django_project/kartoza_theme/page_processors.py

```diff
@@ -0,0 +1,26 @@
+import operator
+from django import forms
+from collections import OrderedDict
+from django.core.exceptions import ObjectDoesNotExist
+from django.http import HttpResponseRedirect
+from mezzanine.pages.page_processors import processor_for
+from mezzanine.pages.models import Page
+from cartridge.shop.models import Category
+from .models import ProductViewData
+
+
+@processor_for(Category)
+def product_view_data(request, page, *args, **kwargs):
+    category = Category.objects.get(pk=page.id)
+    view_data = {}
+    for product in category.product_set.all():
+        try:
+            product_view_data = ProductViewData.objects.filter(product=product).first()
+            if product_view_data:
+                view_data[product.id] = product_view_data
+        except ObjectDoesNotExist:
+            pass
+    sorted_view_data = OrderedDict(
```

Comment:
why not just do this 
```
ProductViewData.objects.filter(product__in=category.product_set.all()).order_by('category_order')
```
?

Reply:
Because I am not as wise and awesome as you are! Thanks, I should remember to always think in queries.